src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplay.vb

```diff
@@ -156,24 +204,52 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+        Private Sub AddSymbolDisplayParts(parts As ArrayBuilder(Of SymbolDisplayPart), c As Char, options As ObjectDisplayOptions)
+            If ObjectDisplay.IsPrintable(c) OrElse Not options.IncludesOption(ObjectDisplayOptions.EscapeNonPrintableCharacters) Then
+                Dim literal = If(options.IncludesOption(ObjectDisplayOptions.UseQuotes), """" & c & """c", c)
+                parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.StringLiteral, Nothing, literal))
                 Return
             End If
 
-            If ObjectDisplay.IsPrintable(c) Then
-                parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.StringLiteral, Nothing, """" & c & """c"))
+            Dim wellKnown = ObjectDisplay.GetWellKnownCharacterName(c)
+            If wellKnown IsNot Nothing Then
+                parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.FieldName, Nothing, wellKnown))
                 Return
             End If
 
             ' non-printable, add "ChrW(codepoint)"
             Dim codepoint = AscW(c)
+            Dim codepointLiteral =
+                If(options.IncludesOption(ObjectDisplayOptions.UseHexadecimalNumbersForCharacters),
+                    "&H" & codepoint.ToString("X"),
+                    codepoint.ToString())
             parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.MethodName, Nothing, "ChrW"))
             parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.Punctuation, Nothing, "("))
-            parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.NumericLiteral, Nothing, "&H" & codepoint.ToString("X")))
+            parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.NumericLiteral, Nothing, codepointLiteral))
             parts.Add(New SymbolDisplayPart(SymbolDisplayPartKind.Punctuation, Nothing, ")"))
         End Sub
+
+        Private Function ToObjectDisplayOptions(quoteStrings As Boolean, useHexadecimalNumbers As Boolean) As ObjectDisplayOptions
+            Dim numberFormat = If(useHexadecimalNumbers, NumericFormat.Hexadecimal, NumericFormat.Decimal)
+            Return ToObjectDisplayOptions(New SymbolDisplayConstantValueOptions(numberFormat, numberFormat, Not quoteStrings))
+        End Function
+
+        Private Function ToObjectDisplayOptions(constantValueOptions As SymbolDisplayConstantValueOptions) As ObjectDisplayOptions
+            Dim options = ObjectDisplayOptions.None
+
+            If constantValueOptions.NumericLiteralFormat = NumericFormat.Hexadecimal Then
+                options = options Or ObjectDisplayOptions.UseHexadecimalNumbers
+            End If
+
+            If constantValueOptions.CharacterValueFormat = NumericFormat.Hexadecimal Then
+                options = options Or ObjectDisplayOptions.UseHexadecimalNumbersForCharacters
+            End If
+
+            If Not constantValueOptions.NoQuotes Then
+                options = options Or ObjectDisplayOptions.UseQuotes Or ObjectDisplayOptions.EscapeNonPrintableCharacters
```

Comment:
interesting that this covers two options.  can you doc why?

Reply:
I don't know why. I am just preserving the existing behavior of the `useQuotes` flag.

Reply:
that's worth a comment :)

Reply:
Why is it worth a comment to state my personal ignorance? I'm sure other people do know.

Reply:
because it helps document to the next person.  Consider someone else coming along and seeing this and going "hrmm.. was this intentional?  why is this going on?"  Perhaps they're doing it to try to fix some bug.  Having a comment explaining that this was just how the previous system worked, and that we're preserving it, can be helpful.

Reply:
But that's the thing. I do *not* know whether it's intentional. I am the "someone else coming along" that you're describing.